src/Internal.AspNetCore.HostingStartup.Sdk/build/Internal.AspNetCore.HostingStartup.Sdk.targets

```diff
@@ -0,0 +1,50 @@
+<!--
+  This file is automatically imported by NuGet into a user's project
+  when it targets a single framework, or in classic (pre 2017) csproj projects.
+-->
+
+<Project>
+
+  <Target Name="GenerateHostingStartupDeps" BeforeTargets="Build" Condition="'@(HostingStartupPackageReference->Count())' != '0'">
+    <PropertyGroup>
+      <_TemplatesDirectory>$(MSBuildThisFileDirectory)..\content\</_TemplatesDirectory>
+      <_DepsOutputDirectory>$(MSBuildProjectDirectory)\$(BaseIntermediateOutputPath)</_DepsOutputDirectory>
+      <_WorkingDirectory>$(_DepsOutputDirectory)\depswork</_WorkingDirectory>
+      <_BasePackagePath>content\additionaldeps\</_BasePackagePath>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_TemplateFiles Include="$(MSBuildThisFileDirectory)..\content\HostingStartup\*.*" />
+      <_HostingStartupPackageReference
+        Include="%(HostingStartupPackageReference.Identity)"
+        Version="%(HostingStartupPackageReference.Version)"
+        WorkingDirectory="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)"
+        Project="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\HostingStartup.csproj"
+        DepsFile="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\HostingStartup.deps.json"
+        DllPath="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\%(HostingStartupPackageReference.Identity).dll"
+        TrimmedDepsFile="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\%(HostingStartupPackageReference.Identity).deps.json"
+        PackagePath="%(HostingStartupPackageReference.Identity)\shared\Microsoft.NETCore.App\$(HostingStartupRuntimeFrameworkVersion)\"
+         />
+    </ItemGroup>
+
+    <RemoveDir Directories="%(_HostingStartupPackageReference.WorkingDirectory)" />
+
+    <Copy SourceFiles="@(_TemplateFiles)" DestinationFolder="%(_HostingStartupPackageReference.WorkingDirectory)" />
+
+    <MSBuild Projects="%(_HostingStartupPackageReference.Project)" Targets="Restore" />
```

Comment:
Restore shouldn't succeed unless you have the properties set `HostingStartupPackageName=%(_HostingStartupPackageReference.Identity);HostingStartupPackageVersion=%(_HostingStartupPackageReference.Version)`, right?